compose/material3/material3/src/skikoMain/kotlin/androidx/compose/material3/KotlinxDatetimeCalendarModel.kt

```diff
@@ -30,7 +30,11 @@ import kotlinx.datetime.toLocalDateTime
 internal class KotlinxDatetimeCalendarModel : CalendarModel {
 
     override val today: CalendarDate
-        get() = Clock.System.now().toCalendarDate(systemTZ)
+        get() = Clock.System.now()
+            .toLocalDateTime(systemTZ)
+            .date
+            .atStartOfDayIn(TimeZone.UTC)
+            .toCalendarDate(TimeZone.UTC)
```

Comment:
Are you sure this is correct?

Maybe use this instead? It seems clearer and closer to Android's implementation:
```
    override val today: CalendarDate
        get() {
            val localDate = Clock.System.now().toLocalDateTime(systemTZ)
            return CalendarDate(
                year = localDate.year,
                month = localDate.monthNumber,
                dayOfMonth = localDate.dayOfMonth,
                utcTimeMillis = localDate.date
                    .atTime(hour = 0, minute = 0)
                    .toInstant(TimeZone.UTC).toEpochMilliseconds()
            )
        }
```
(just a suggestion; times are hard and it's hard to be sure of anything).

Reply:
This indeed looks cleaner. I’ll check if it produces the same result as legacy model and let you known